unit/test-suffix-si-parse.c

```diff
@@ -11,51 +13,61 @@
 
 static int test_rc;
 
-static void check_num(const char *val, int lbas, __u64 exp, __u64 num)
+static void check_num(const char *val, __u64 exp, __u64 num)
 {
 	if (exp == num)
 		return;
 
-	printf("ERROR: printing {%s} (lbas %d), got '%llu', expected '%llu'\n",
-	       val, lbas, (unsigned long long)num, (unsigned long long)exp);
+	printf("ERROR: printing {%s},  got '%llu', expected '%llu'\n",
+	       val, (unsigned long long)num, (unsigned long long)exp);
 
 	test_rc = 1;
 }
 
 struct tonum_test {
 	const char *val;
-	int lbas;
-	const __u64 exp;
+	const uint64_t exp;
+	int ret;
 };
 
 static struct tonum_test tonum_tests[] = {
-	{ "11995709440", 512, 11995709440 },
-	{ "1199570940", 512, 1199570940 },
-	{ "6.14T", 512, 11992187500 },
-	{ "6.14T", 520, 11807692307 },
-	{ "6.14T", 4096, 1499023437 },
-	{ "6.14", 512, 0 },
-	{ "6.14#", 512, 0 },
+	{ "11995709440", 11995709440, 0 },
+	{ "1199570940", 1199570940, 0},
+	{ "234.567M", 234567000, 0 },
```

Comment:
I think that the val "234.567MM" is parsed as "234.567M". If so is it expected?

Reply:
Good point, this should be handled as well.

Reply:
Updated the check and also the tests case.